sdk/tests/js/tests/tex-image-and-sub-image-3d-with-image.js

```diff
@@ -248,10 +270,111 @@ function generateTest(internalFormat, pixelFormat, pixelType, prologue, resource
         }
     }
 
+    function runMipMapTest(image) {
+
+        var width = image.width;
+        var sourceSubRectangle = [0, 0, width, width];
+        var depth = 4;
+        var useTexSubImage3D = false;
+        var useMinMax = false;
+        var tolerance = 3; // Since using non-0/1 values, on various formats the results won't be exact
+
+        function runOneIteration(bindingTarget, rTextureCoord, lod, color, program)
+        {
+            debug('Testing ' + (useTexSubImage3D ? 'texSubImage3D' : 'texImage3D') +
+                  ' with bindingTarget=' +
+                  (bindingTarget == gl.TEXTURE_3D ? 'TEXTURE_3D' : 'TEXTURE_2D_ARRAY') + ' with mipmaps from ' +
+                  image.width + 'x' + image.height + ' with r=' + rTextureCoord + ', lod=' + lod);
+
+            uploadImageToTexture(image, useTexSubImage3D, false, bindingTarget, depth, sourceSubRectangle, undefined, true);
+
+            var rCoordLocation = gl.getUniformLocation(program, 'uRCoord');
+            if (!rCoordLocation) {
+                testFailed('Shader incorrectly set up; couldn\'t find uRCoord uniform');
+                return;
+            }
+            gl.uniform1f(rCoordLocation, rTextureCoord);
+
+            if (useMinMax) {
+                // Note: this seems like it should work, but fails on uint textures,
+                //   bug in test / test author, or bug in implementations?
+                gl.texParameterf(bindingTarget, gl.TEXTURE_MIN_LOD, lod);
```

Comment:
Using `TEXTURE_MIN/MAX_LOD` here works in all tests except for the unsigned integer formats (where it appears to sample from the wrong level).  I'm not sure if that means there's something I don't understand, or if I've stumbled upon a bug.  I'd be happy to change this code to also test the alternative way to choose which lod levels are sampled from, or to remove this (currently unreachable) branch if desired.